src/AnnotationDialog.js

```diff
@@ -569,106 +546,38 @@ class AnnotationDialog extends EventEmitter {
-            minute: '2-digit'
-        });
-        const textEl = util.createCommentTextNode(annotation.text);
-
-        const annotationEl = document.createElement('div');
-        annotationEl.classList.add(CLASS_COMMENT);
-        annotationEl.setAttribute('data-annotation-id', annotation.annotationID);
-
+    renderAnnotations() {
         const annotationContainerEl = this.dialogEl.querySelector(`.${CLASS_COMMENTS_CONTAINER}`);
-        annotationContainerEl.appendChild(annotationEl);
-
-        // Avatar
-        const avatarEl = document.createElement('div');
-        avatarEl.classList.add(constants.CLASS_PROFILE_IMG_CONTAINER);
-        avatarEl.innerHTML = avatarHtml;
-        annotationEl.appendChild(avatarEl);
-
-        // Creator namate & date
-        const profileContainerEl = document.createElement('div');
-        profileContainerEl.classList.add(constants.CLASS_PROFILE_CONTAINER);
-        annotationEl.appendChild(profileContainerEl);
-
-        const userNameEl = document.createElement('div');
-        userNameEl.classList.add(constants.CLASS_USER_NAME);
-        userNameEl.textContent = userName;
-        profileContainerEl.appendChild(userNameEl);
-
-        const createdEl = document.createElement('div');
-        createdEl.classList.add(constants.CLASS_COMMENT_DATE);
-        createdEl.textContent = created;
-        profileContainerEl.appendChild(createdEl);
-
-        // Comment
-        const commentTextEl = document.createElement('div');
-        commentTextEl.appendChild(textEl);
-        annotationEl.appendChild(commentTextEl);
-
-        // Delete button
-        if (!annotation.permissions.can_delete) {
-            return;
-        }
-
-        const deleteBtn = util.generateBtn(
-            [constants.CLASS_BUTTON_PLAIN, CLASS_BUTTON_DELETE_COMMENT],
-            this.localized.deleteButton,
-            ICON_DELETE,
-            constants.DATA_TYPE_DELETE
-        );
-        annotationEl.appendChild(deleteBtn);
-
-        const deleteConfirmEl = document.createElement('div');
-        deleteConfirmEl.classList.add(CLASS_DELETE_CONFIRMATION);
-        deleteConfirmEl.classList.add(constants.CLASS_HIDDEN);
-        annotationEl.appendChild(deleteConfirmEl);
-
-        const confirmMsgEl = document.createElement('div');
-        confirmMsgEl.classList.add(constants.CLASS_DELETE_CONFIRM_MESSAGE);
-        confirmMsgEl.textContent = this.localized.deleteConfirmation;
-        deleteConfirmEl.appendChild(confirmMsgEl);
-
-        const deleteBtnsEl = document.createElement('div');
-        deleteBtnsEl.classList.add(constants.CLASS_BUTTON_CONTAINER);
-        deleteConfirmEl.appendChild(deleteBtnsEl);
-
-        const cancelDeleteBtn = util.generateBtn(
-            [constants.CLASS_BUTTON, CLASS_CANCEL_DELETE],
-            this.localized.cancelButton,
-            this.localized.cancelButton,
-            constants.DATA_TYPE_CANCEL_DELETE
+        const locale = this.locale.substr(0, this.locale.indexOf('-'));
```

Comment:
Is there a plan to support full localization beyond just the main language?

Reply:
Hmm so looks like this is how it's supported right now in react-intl https://github.com/yahoo/react-intl/wiki/Components#intlprovider

Reply:
Which seems like we don't need to parse out the language then, and can pass in the locale directly and let `react-intl` do it's magic then?

Reply:
https://github.com/box/box-annotations/pull/216